orttraining/orttraining/models/runner/training_runner.cc

```diff
@@ -150,11 +176,58 @@ Status TrainingRunner::Initialize() {
 
   opt_graph_outputs_ = config_result.opt_config_result.value().output_key_to_graph_output_name;
 
-  // Expose all fetches as graph outputs
-  VectorString fetch_names = params_.fetch_names;
+  // Retrieve pipeline information from configuration result.
+  VectorString fetch_names;
+  if (params_.num_pipeline_stages > 1) {
+    fetch_names = config_result.pipeline_config_result.value().fetch_names;
+    // Exposes forward waited event tensor ID name to TrainingRunner.
+    // It's an input of a graph.
+    pipeline_context_.forward_waited_event_name = config_result.pipeline_config_result.value().forward_waited_event_name;
+    // Exposes forward recorded event tensor ID name to TrainingRunner.
+    // It's an input of a graph.
+    pipeline_context_.forward_recorded_event_name = config_result.pipeline_config_result.value().forward_recorded_event_name;
+    // Exposes backward waited event tensor ID name to TrainingRunner.
+    // It's an input of a graph.
+    pipeline_context_.backward_waited_event_name = config_result.pipeline_config_result.value().backward_waited_event_name;
```

Comment:
consider adding some assertion for these value? For example, if it is the last stage, these two strings should be empty?

Reply:
I personally don't want to add too many checks here because the final `shape` if pipeline is not determined yet. For example, we are adding two Record's and two Wait's per pipeline stage. Maybe add later once the design is finalized?